astrodb_utils/sources.py

```diff
@@ -100,33 +102,50 @@ def find_source_in_db(
     # if still no matches, try spatial search using coordinates, if provided
     if len(db_name_matches) == 0 and ra and dec:
         location = SkyCoord(ra, dec, frame="icrs", unit="deg")
-        radius = u.Quantity(search_radius, unit="arcsec")
         logger.debug(
             f"{source}: Trying coord in database search around "
             f"{location.ra.degree}, {location.dec}"
         )
         db_name_matches = db.query_region(
-            location, radius=radius, ra_col=ra_col_name, dec_col=dec_col_name
+            location, radius=search_radius, ra_col=ra_col_name, dec_col=dec_col_name
         )
 
     # If still no matches, try to get the coords from SIMBAD using source name
     if len(db_name_matches) == 0 and use_simbad:
         simbad_skycoord = coords_from_simbad(source)
         # Search database around that coordinate
         if simbad_skycoord is not None:
-            radius = u.Quantity(search_radius, unit="arcsec")
             msg2 = (
-                f"Finding sources around {simbad_skycoord} with radius {radius} "
+                f"Finding sources around {simbad_skycoord} with radius {search_radius} "
                 f"using ra_col_name: {ra_col_name}, dec_col_name: {dec_col_name}"
             )
             logger.debug(msg2)
             db_name_matches = db.query_region(
-                simbad_skycoord, radius=radius, ra_col=ra_col_name, dec_col=dec_col_name
+                simbad_skycoord, radius=search_radius, ra_col=ra_col_name, dec_col=dec_col_name
             )
 
     if len(db_name_matches) == 1:
         db_names = db_name_matches["source"].tolist()
         logger.debug(f"One match found for {source}: {db_names[0]}")
+
+        # check coords of the match make sense
+        coord_check = (
+            ra
+            and dec
+            and u.isclose(ra*u.degree, db_name_matches[ra_col_name][0]*u.degree,atol=search_radius)
+            and np.isclose(dec*u.degree, db_name_matches[dec_col_name][0]*u.degree, atol=search_radius)
```

Comment:
You are using `u.isclose` from astropy.units for RA and `np.isclose` from numpy for dec. I didn't know that astropy.units had this, but do you want to be consistent and use the same for both?

Reply:
Yes!  that was an oversight.  The u.isclose thing is cool!